### ISSUE TITLE
Tidying `anova()` output from `lme4` models

### ISSUE THREAD

**Issue by AmeliaMN** [tidymodels/broom#1114]
## The problem

I'm re-running some code that worked before I updated R/all my packages. I'm getting an error from `broom` related to duplicated names in the `anova()` output from `lme4` models. 

## Reproducible example

``` r
  library(tidyverse)
  library(lme4)
#> Loading required package: Matrix
#> 
#> Attaching package: 'Matrix'
#> The following objects are masked from 'package:tidyr':
#> 
#>     expand, pack, unpack
library(broom)
fm1 <- lmer(Reaction ~ Days + (Days | Subject), sleepstudy)
fm2 <- lmer(Reaction ~ Days + (Days || Subject), sleepstudy)
anova(fm1, fm2) %>%
  tidy()
#> refitting model(s) with ML (instead of REML)
#> Warning in tidy.anova(.): The following column names in ANOVA output were not
#> recognized or transformed: npar
#> Error in `mutate()`:
#> ! Can't transform a data frame with duplicate names.
```

<sup>Created on 2022-08-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.1 (2022-06-23)
#>  os       macOS Monterey 12.2.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       America/Chicago
#>  date     2022-08-22
#>  pandoc   2.18 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package       * version    date (UTC) lib source
#>  assertthat      0.2.1      2019-03-21 [1] CRAN (R 4.2.0)
#>  backports       1.4.1      2021-12-13 [1] CRAN (R 4.2.0)
#>  boot            1.3-28     2021-05-03 [1] CRAN (R 4.2.1)
#>  broom         * 1.0.0.9000 2022-08-22 [1] Github (tidymodels/broom@1f017a5)
#>  cellranger      1.1.0      2016-07-27 [1] CRAN (R 4.2.0)
#>  cli             3.3.0      2022-04-25 [1] CRAN (R 4.2.0)
#>  colorspace      2.0-3      2022-02-21 [1] CRAN (R 4.2.0)
#>  crayon          1.5.1      2022-03-26 [1] CRAN (R 4.2.0)
#>  DBI             1.1.3      2022-06-18 [1] CRAN (R 4.2.0)
#>  dbplyr          2.2.1      2022-06-27 [1] CRAN (R 4.2.0)
#>  digest          0.6.29     2021-12-01 [1] CRAN (R 4.2.0)
#>  dplyr         * 1.0.9      2022-04-28 [1] CRAN (R 4.2.0)
#>  ellipsis        0.3.2      2021-04-29 [1] CRAN (R 4.2.0)
#>  evaluate        0.16       2022-08-09 [1] CRAN (R 4.2.0)
#>  fansi           1.0.3      2022-03-24 [1] CRAN (R 4.2.0)
#>  fastmap         1.1.0      2021-01-25 [1] CRAN (R 4.2.0)
#>  forcats       * 0.5.1      2021-01-27 [1] CRAN (R 4.2.0)
#>  fs              1.5.2      2021-12-08 [1] CRAN (R 4.2.0)
#>  gargle          1.2.0      2021-07-02 [1] CRAN (R 4.2.0)
#>  generics        0.1.3      2022-07-05 [1] CRAN (R 4.2.0)
#>  ggplot2       * 3.3.6      2022-05-03 [1] CRAN (R 4.2.0)
#>  glue            1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
#>  googledrive     2.0.0      2021-07-08 [1] CRAN (R 4.2.0)
#>  googlesheets4   1.0.1      2022-08-13 [1] CRAN (R 4.2.0)
#>  gtable          0.3.0      2019-03-25 [1] CRAN (R 4.2.0)
#>  haven           2.5.0      2022-04-15 [1] CRAN (R 4.2.0)
#>  highr           0.9        2021-04-16 [1] CRAN (R 4.2.0)
#>  hms             1.1.1      2021-09-26 [1] CRAN (R 4.2.0)
#>  htmltools       0.5.3      2022-07-18 [1] CRAN (R 4.2.0)
#>  httr            1.4.3      2022-05-04 [1] CRAN (R 4.2.0)
#>  jsonlite        1.8.0      2022-02-22 [1] CRAN (R 4.2.0)
#>  knitr           1.39       2022-04-26 [1] CRAN (R 4.2.0)
#>  lattice         0.20-45    2021-09-22 [1] CRAN (R 4.2.1)
#>  lifecycle       1.0.1      2021-09-24 [1] CRAN (R 4.2.0)
#>  lme4          * 1.1-30     2022-07-08 [1] CRAN (R 4.2.0)
#>  lubridate       1.8.0      2021-10-07 [1] CRAN (R 4.2.0)
#>  magrittr        2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
#>  MASS            7.3-57     2022-04-22 [1] CRAN (R 4.2.1)
#>  Matrix        * 1.4-1      2022-03-23 [1] CRAN (R 4.2.1)
#>  minqa           1.2.4      2014-10-09 [1] CRAN (R 4.2.0)
#>  modelr          0.1.8      2020-05-19 [1] CRAN (R 4.2.0)
#>  munsell         0.5.0      2018-06-12 [1] CRAN (R 4.2.0)
#>  nlme            3.1-157    2022-03-25 [1] CRAN (R 4.2.1)
#>  nloptr          2.0.3      2022-05-26 [1] CRAN (R 4.2.0)
#>  pillar          1.8.1      2022-08-19 [1] CRAN (R 4.2.0)
#>  pkgconfig       2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
#>  purrr         * 0.3.4      2020-04-17 [1] CRAN (R 4.2.0)
#>  R.cache         0.16.0     2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3     1.8.2      2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo            1.25.0     2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils         2.12.0     2022-06-28 [1] CRAN (R 4.2.0)
#>  R6              2.5.1      2021-08-19 [1] CRAN (R 4.2.0)
#>  Rcpp            1.0.9      2022-07-08 [1] CRAN (R 4.2.0)
#>  readr         * 2.1.2      2022-01-30 [1] CRAN (R 4.2.0)
#>  readxl          1.4.0      2022-03-28 [1] CRAN (R 4.2.0)
#>  reprex          2.0.1      2021-08-05 [1] CRAN (R 4.2.0)
#>  rlang           1.0.4      2022-07-12 [1] CRAN (R 4.2.0)
#>  rmarkdown       2.14       2022-04-25 [1] CRAN (R 4.2.0)
#>  rstudioapi      0.13       2020-11-12 [1] CRAN (R 4.2.0)
#>  rvest           1.0.2      2021-10-16 [1] CRAN (R 4.2.0)
#>  scales          1.2.1      2022-08-20 [1] CRAN (R 4.2.0)
#>  sessioninfo     1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
#>  stringi         1.7.8      2022-07-11 [1] CRAN (R 4.2.0)
#>  stringr       * 1.4.1      2022-08-20 [1] CRAN (R 4.2.0)
#>  styler          1.7.0      2022-03-13 [1] CRAN (R 4.2.0)
#>  tibble        * 3.1.8      2022-07-22 [1] CRAN (R 4.2.0)
#>  tidyr         * 1.2.0      2022-02-01 [1] CRAN (R 4.2.0)
#>  tidyselect      1.1.2      2022-02-21 [1] CRAN (R 4.2.0)
#>  tidyverse     * 1.3.2      2022-07-18 [1] CRAN (R 4.2.0)
#>  tzdb            0.3.0      2022-03-28 [1] CRAN (R 4.2.0)
#>  utf8            1.2.2      2021-07-24 [1] CRAN (R 4.2.0)
#>  vctrs           0.4.1      2022-04-13 [1] CRAN (R 4.2.0)
#>  withr           2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun            0.32       2022-08-10 [1] CRAN (R 4.2.0)
#>  xml2            1.3.3      2021-11-30 [1] CRAN (R 4.2.0)
#>  yaml            2.3.5      2022-02-21 [1] CRAN (R 4.2.0)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

I suspect `lme4` has changed their default printing for `anova()`, because the output looks different than a standard `anova()` table. 

``` r
  library(tidyverse)
  library(lme4)
#> Loading required package: Matrix
#> 
#> Attaching package: 'Matrix'
#> The following objects are masked from 'package:tidyr':
#> 
#>     expand, pack, unpack
library(broom)
fm1 <- lmer(Reaction ~ Days + (Days | Subject), sleepstudy)
fm2 <- lmer(Reaction ~ Days + (Days || Subject), sleepstudy)
anova(fm1, fm2)
#> refitting model(s) with ML (instead of REML)
#> Data: sleepstudy
#> Models:
#> fm2: Reaction ~ Days + ((1 | Subject) + (0 + Days | Subject))
#> fm1: Reaction ~ Days + (Days | Subject)
#>     npar    AIC    BIC  logLik deviance  Chisq Df Pr(>Chisq)
#> fm2    5 1762.0 1778.0 -876.00   1752.0                     
#> fm1    6 1763.9 1783.1 -875.97   1751.9 0.0639  1     0.8004
```

<sup>Created on 2022-08-22 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bbolker** [comment]
As stated in the issue linked above, my best guess is that this caused by the recent change to add a `terms` column to the output, specifically [these lines](https://github.com/tidymodels/broom/commit/358df2d39f7adcc884749084a3f7e7820505bc15#diff-3f973b67aae9cb03ab4abe2e4f4fea231ddf84999e04c55c7d45a89f8b00779dR130-R136) which make some assumptions about the structure of the header.  I think the regexps can be relaxed here, will take a crack ...

**Comment by simonpcouch** [comment]
Thank you for the issue, @AmeliaMN. :)

Yes, @bbolker, happen to be poking around with the same code right now! Was indeed introduced in #1106. I'll wait to push my changes and will look out for a PR.

**Comment by simonpcouch** [comment]
Fixed in #1115! These changes will make it to CRAN in the next few days. Thanks again. :)

``` r
library(tidyverse)
library(lme4)
#> Loading required package: Matrix
#> 
#> Attaching package: 'Matrix'
#> The following objects are masked from 'package:tidyr':
#> 
#>     expand, pack, unpack
library(broom)
  
fm1 <- lmer(Reaction ~ Days + (Days | Subject), sleepstudy)
fm2 <- lmer(Reaction ~ Days + (Days || Subject), sleepstudy)

anova(fm1, fm2) %>%
  tidy()
#> refitting model(s) with ML (instead of REML)
#> # A tibble: 2 × 9
#>   term   npar   AIC   BIC logLik deviance statistic    df p.value
#>   <chr> <dbl> <dbl> <dbl>  <dbl>    <dbl>     <dbl> <dbl>   <dbl>
#> 1 fm2       5 1762. 1778.  -876.    1752.   NA         NA  NA    
#> 2 fm1       6 1764. 1783.  -876.    1752.    0.0639     1   0.800
```

<sup>Created on 2022-08-29 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>
